datadog/resource_datadog_dashboard.go

```diff
@@ -5380,6 +5473,20 @@ func getTimeseriesRequestSchema() map[string]*schema.Schema {
 	}
 }
 
+func buildDatadogScatterplotFormula(data map[string]interface{}) datadogV1.ScatterplotWidgetFormula {
+	formula := datadogV1.ScatterplotWidgetFormula{}
+	if formulaExpression, ok := data["formula_expression"].(string); ok && len(formulaExpression) != 0 {
+		formula.SetFormula(formulaExpression)
+	}
+	if alias, ok := data["alias"].(string); ok && len(alias) != 0 {
+		formula.SetAlias(alias)
+	}
+	if dimension, ok := data["dimension"].(string); ok && len(dimension) != 0 {
+		formula.SetDimension(datadogV1.ScatterplotDimension(dimension))
+	}
```

Comment:
`alias` and `dimension` are optional, so you should test for existence before casting to `string`, otherwise that might panic.

Reply:
Tested this locally and not setting `alias` didn't blow anything up (and it properly stored the state as an empty string). I've marked `dimension` as required now so this is safe.